app/views/posts/index.html.erb

```diff
@@ -1 +1,39 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<ul class="users_container">
+ <%= render "shared/user_profile", user: @user %>
+  <% @posts.each_with_index do |post, index| %>
+    <%= link_to user_post_path(@user.id, post.id) do %>
+      <li class="user_all_posts">
+        <div class="user_bio">
+          <h3>Post#<%= index+1 %></h3>
+          <p><%= post.Text.size > 100 ? "#{post.Text[0..100]} ..." : post.Text %></p>
+          <div class="user_recent_posts">
+            <span>
+              Comments: <%= post.comments.count %>,
+            </span>
+            <span>
+              Likes: <%= post.Likes_counter.nil? ? '0' : post.Likes_counter %>
+            </span>
+          </div>
+        </div>
+
+        <div class="user_post_comments">
+          <% if post.comments.empty? %>
+            <div class="user_post_comment">
+              <p>No comments</p>
+            </div>
+          <% else %>
+            <% post.comments[1..5].each_with_index do |comment, index| %>
+              <div class="user_post_comment">
+                <h3><%= index + 1 %>. <%= comment.user.Name %></h3>
+                <p><%= comment.Text %></p>
+              </div>
+            <% end %>
+          <% end %>
+        </div>
```

Comment:
- [ ] Great job on your implementation so far! 🚀 However, there are a few minor issues that need some attention to ensure your code functions as expected. 

   - It appears that the data from the database isn't displaying correctly in the UI. This issue arises from the way you're referencing your data column names. I've noticed that you're using capital letters at the beginning of the column names, but they should match how they are saved in the database, typically with lowercase letters. 

   For example, in the highlighted code, you can access the `text` from your database like this:

   ```ruby
      <p><%= post.text.size > 100 ? "#{post.text[0..100]} ..." : post.text %></p>
   ``` 

   Instead of:

   ```ruby
       <p><%= post.Text.size > 100 ? "#{post.Text[0..100]} ..." : post.Text %></p>
   ``` 

   - Kindly make sure to make this correction across your entire codebase. Some of the column names you need to correct in this file includes: `name`, `text`, and `likes_counter`.

Happy coding! 😊👨‍💻🚀